code/modules/mob/living/simple_animal/hostile/hivebot.dm

```diff
@@ -31,85 +68,207 @@
+	projectilesound = 'sound/weapons/taser2.ogg'
+	projectiletype = /obj/item/projectile/beam/hivebotdischarge
+	wander = 0
 	status_flags = 0
-	anchored = 1
-	stop_automated_movement = 1
-	var/bot_type = NORMAL // type of bot, 1 is normal, 2 is ranged, 3 is rapid rnaged.
-	var/bot_amt = 10
-	var/spawn_delay = 600
+	faction = "hivebot"
+	ranged = 1
+	rapid = 1
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	minbodytemp = 0
+	speed = -10
+	see_in_dark = 8
+	smart = 1
+	destroy_surroundings = 0
+	var/bot_type = NORMAL // type of bot, 1 is normal, 2 is ranged, 3 is rapid ranged
+	var/bot_amt = 48 //Number of total bots that are spawned before the beacon disappears completely.
+	var/max_bots = 16 //Number of bots linked to this beacon specifically that can exist, before spawning more is halted.
+	var/list/linked_bots = list()
+	var/latest_child = null
+	var/existing_bots = 0
+	var/spawn_delay = 300
+	var/activated = 0
+	var/snoozing = 0 //If set to 1, it will be prevented from spawning bots, unless it spots an enemy.
+	attack_emote = "focuses on"
+
+/mob/living/simple_animal/hostile/hivebotbeacon/toxic
+	projectiletype = /obj/item/projectile/beam/hivebotbeam
+	projectilesound = 'sound/weapons/laser3.ogg'
+	rapid = 0
 
-/mob/living/simple_animal/hostile/hivebot/tele/New()
+/mob/living/simple_animal/hostile/hivebotbeacon/incendiary
+	projectiletype = /obj/item/projectile/beam/hivebotincendiary
+	projectilesound = 'sound/weapons/plasma_cutter.ogg'
+	rapid = 0
+
+/mob/living/simple_animal/hostile/hivebotbeacon/New()
 	..()
-	var/datum/effect/effect/system/smoke_spread/smoke = new /datum/effect/effect/system/smoke_spread()
-	smoke.set_up(5, 0, src.loc)
-	smoke.start()
-	visible_message("<span class='danger'>\The [src] warps in!</span>")
+	var/datum/effect/effect/system/smoke_spread/S = new /datum/effect/effect/system/smoke_spread()
+	S.set_up(5, 0, src.loc)
+	S.start()
+	icon_state = "hivebotbeacon_off"
+	visible_message("<span class='danger'>\ [src] warps in!</span>")
 	playsound(src.loc, 'sound/effects/EMPulse.ogg', 25, 1)
 
-/mob/living/simple_animal/hostile/hivebot/tele/proc/warpbots()
+/mob/living/simple_animal/hostile/hivebotbeacon/death()
+	..(null,"blows apart and erupts in a cloud of noxious smoke!")
+	var/datum/effect/effect/system/smoke_spread/S = new /datum/effect/effect/system/smoke_spread
+	S.set_up(5, 0, src.loc)
+	S.start()
+	new /obj/effect/decal/cleanable/greenglow(src.loc)
+	var/T = get_turf(src)
+	new /obj/effect/gibspawner/robot(T)
+	spark(T, 3, alldirs)
+	for (var/mob/living/simple_animal/hostile/hivebot/latest_child in linked_bots)
+		latest_child.linked_parent = null
+	linked_bots.Cut()
+	qdel(src)
+	return
+
+/mob/living/simple_animal/hostile/hivebotbeacon/think()
+	. =..()
+	if(stance != HOSTILE_STANCE_IDLE && activated == 0)
+		visible_message("<span class='warning'>\ [src] suddenly activates!</span>")
+		icon_state = "hivebotbeacon_raising"
+		spawn(16)

```

Comment:
This is ineffectual. `spawn()` guards a block of code, but there is no block below it.
Also, use a timer instead of spawn.

```dm
/mob/living/simple_animal/hostile/hivebotbeacon/proc/reset_activation()
	icon_state = "hivebotbeacon_active"
	activated = 1
	snoozing = 0
```

Use it like this:
```dm
addtimer(CALLBACK(src, .proc/reset_activation), 16)
```